casepro/msgs/views.py

```diff
@@ -186,7 +187,23 @@ def get_context_data(self, **kwargs):
             user = self.request.user
             page = int(self.request.GET.get('page', 1))
 
+            context['user_id'] = user.id
+
             search = self.derive_search()
+
+            # this is a refresh of messages
+            if self.request.GET.get('last_refresh', None):
+                new_messages = Message.search(org, user, search)
+
+                search['last_refresh'] = self.request.GET['last_refresh']
+                del search['after']
```

Comment:
@erikh360 Thanks for explaining. Sorry, I think my question wasn't clear. I agree that it doesn't make sense return messages matching `after` filter _and_ the `last_refresh` filter. What I'm wondering is why we don't support returning messages matching `after` filter _or_ `last_refresh` filter.

After reviewing a bit more, I now see we are actually doing this, but that [this logic lives in the view](https://github.com/praekelt/casepro/blob/118ae768757fb26634c58b709d5f943b532a94e3/casepro/msgs/views.py#L196-L200), and achieves this by calling `Message.search` twice, once without `last_refresh`, and once with it.

My gut feel is that this logic should live in `Message.search`, but if that is not possible, or more work than it is worth, I'm alright with the current approach.